ci-operator/config/openshift/hypershift/openshift-hypershift-main.yaml

```diff
@@ -735,6 +759,25 @@ tests:
         requests:
           cpu: "3"
           memory: 600Mi
+- as: e2e-agent
+  optional: true
+  run_if_changed: ^(api/.*|hypershift-operator/controllers/hostedcluster/internal/platform/agent/.*)$
+  steps:
+    cluster_profile: packet-assisted
+    dependencies:
+      HYPERSHIFT_IMAGE: pipeline:hypershift
+    env:
+      ASSISTED_CONFIG: |
+        KUBE_API=yes
+        ENABLE_KUBE_API=true
+        MINIKUBE_DISK_SIZE=70g
+        MINIKUBE_RAM_MB=10240
+        TEST=./src/tests/test_kube_api.py
+        TEST_FUNC=test_capi_provider
+        MAKEFILE_TARGET="create_full_environment run deploy_capi_env test"
+      GATHER_CAPI_LOGS: "true"
+      TEST_SUITE: minimal
```

Comment:
doesn't this need to have a 
```
commands: |-
        bin/test-e2e \ ....
```

like in aws to run the e2e hypershift e2e?

Reply:
The test suite is a set of the Kubernetes conformance tests that run against the resulting HyperShift cluster.
I'm not sure what is tested in `hypershift e2e` but according to the args it gets (pull secret, aws-region, release-image) it seems more related to the HyperShift operator functionality (install cluster/ scaleup/down).
In order to run such tests for the agent platform a lot of infrastructure setup is required, this setup is provided by the baremetalds-assisted workflow.

Reply:
ok thanks, I see. Then I don't think this should be called e2e-agent as e2e-aws/platform it's being used to run hypershift e2e tests. e2e-aws/platform validates expectations against hostedCluster and nodePool management, it does not need to be in this PR but the agent platform should run those as well.

Reply:
I'll add the same coverage for the agent platform.
For now, I'd like to start with this job configuration to allow us to make changes to the agent platform and get some coverage on the PRs.